_posts/en/newsletters/2019-05-28-newsletter.md

```diff
@@ -229,73 +229,69 @@ would only require them to submit a Bitcoin address and an email address
-for more than a dozen or so participants.  However, a more clever
-construction could probably avoid that problem by allowing multiple
-participants to leave the tree together at the same point (e.g., after 5
-individual withdrawals, the remaining participants must leave in groups of
-five).
-
-Again, the key benefit here is the ability of COSHV to allow the
-construction of this complex multi-transaction tree without any
-interaction from the receivers after receiving each participant's own
-desired withdrawal address.  It's possible to construct the same tree
-using just the pre-signed transactions we saw in the first sub-section
-above, but the same combinatorial blowup would exist there---and it'd be
-made worse by the fact that each possible combination would need to be
-interactively signed by all participants before the setup transaction
-could be broadcast.  This could easily require the signing of millions
-of combinations by each participant.
+money from the covenant while also allowing most the other receivers to
+keep their money offchain for the moment.
+
+For example, let's imagine Alice paid each of our ten receivers 1 BTC
+using a tree of outputs where each output paid two other outputs until
+an individual's money was withdrawn from the covenant.  Or, at each
+level, an alternative Taproot option would allow paying out all
+remaining outputs at that point.  A complete tree might for ten
+participants might look like this:
+
+![Example output tree for ten participants](/img/posts/2019-05-tx-tree-5.png)
+{:.center}
+
+If receiver Bob wants to individually withdraw now, he could naively
+follow the whole path until he received his output, allowing him to
+create only five outputs instead of the ten he'd need to create if
+everyone had to exit at the same time.
+
+![Example output tree to allow participant Bob to exit](/img/posts/2019-05-tx-tree-6.png)
+{:.center}
+
+Since Bob is the first participant to withdraw, he has to create more
+outputs than everyone else, about `log2(number_of_participants)`.
+Anyone else exiting after Bob would need to create even fewer outputs as
+they could just start spending from one of the intermediate COSHV
+outputs Bob had created.  Some people (like Charlie in the above
+example) would get their exit outputs as a byproduct of another person's
+exit.  Overall, the total number of outputs that need to be created
+would be about double the number of participants.  However, this could
+be just the one output per participant from the previous subsection if
+one of the taproot leaves at the top level also allowed just sending the
+regular batched payment transaction---allowing everyone to exit together
+efficiently.
+
+This balanced tree construction provides a good generic starting part for
+allowing individuals to exit from the covenant without having to pay all
+the exit costs themselves.  When applying COSHV to specific cases where
+it's known some exits are more likely than others, it's possible to
```

Comment:
With COSHV all exits are certain (p=1) so none are more likely than others; I think the difference is some are more _urgent_ than others. The outcome's the same -- since they're more urgent they're more likely to hit during high fees, so a shorter path saves fees.